draft-ietf-ohai-ohttp.md

```diff
@@ -684,11 +684,19 @@ indicating the content type, and the encapsulated response as the response
 content.  As with requests, additional fields MAY be used to convey information
 that does not reveal information about the encapsulated response.
 
-An Oblivious Gateway Resource that does not receive a response can itself
-generate a response with an appropriate error status code (such as 504 (Gateway
-Timeout); see {{Section 15.6.5 of HTTP}}), which is then encapsulated in the
+An Oblivious Gateway Resource that fails to process the decapsulated request
+or does not receive a response from the Target Resource can itself generate a
+response with an appropriate error status code (such as 400 (Bad Request) or
+504 (Gateway Timeout); see {{Section 15.5.1 of HTTP}} and
+{{Section 15.6.5 of HTTP}}, respectively), which is then encapsulated in the
 same way as a successful response.
 
+An Oblivious Gateway Resource which fails to decapsute as described in {{request}}
+because the Encapsulated Request key ID is invalid SHOULD send a 401 status code.
```

Comment:
I don't think that 401 is appropriate here.  I would avoid being too specific and say 4xx for any errors with decapsulation.

Reply:
It's been useful in practice to distinguish failures due to bad configuration from other failures. What would you suggest we do to allow that?

Reply:
Yeah 4XX is probably better here. If the key implies client “authentication” then 401 makes sense, but that’s not really a general interpretation.

Reply:
4xx doesn't seem sufficient here since that would admit 400, which wouldn't allow the client to distinguish configuration mismatch from other failures.

Reply:
Welcome to HTTP.  The usual way we deal with that is that we include details in the body of the response.

Reply:
Sure, but we're defining a protocol here, and I think it's important that there be a very clear signal to clients in this case. Letting implementations do this any way they want using the body is a bad outcome.

Reply:
I'll just crib what ACME did for errors here. We probably should have done that in the first place.

Reply:
Done. Please have a look.